hummingbot/market/kraken/kraken_market.pyx

```diff
@@ -0,0 +1,1077 @@
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+                await safe_gather(
+                    self._update_balances(),
+                    self._update_order_status(),
+                )
+                self._last_pull_timestamp = self._current_timestamp
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network("Unexpected error while fetching account updates.", exc_info=True,
+                                      app_warning_msg="Could not fetch account updates from Kraken. "
+                                                      "Check API key and network connection.")
+                await asyncio.sleep(0.5)
+
+    async def _trading_rules_polling_loop(self):
+        while True:
+            try:
+                await safe_gather(
+                    self._update_trading_rules(),
+                )
+                await asyncio.sleep(60)
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network("Unexpected error while fetching trading rules.", exc_info=True,
+                                      app_warning_msg="Could not fetch new trading rules from Kraken. "
+                                                      "Check network connection.")
+                await asyncio.sleep(0.5)
+
+    @property
+    def status_dict(self) -> Dict[str, bool]:
+        return {
+            "order_books_initialized": self._order_book_tracker.ready,
+            "account_balance": len(self._account_balances) > 0 if self._trading_required else True,
+            "trading_rule_initialized": len(self._trading_rules) > 0,
+        }
+
+    @property
+    def ready(self) -> bool:
+        return all(self.status_dict.values())
+
+    cdef c_start(self, Clock clock, double timestamp):
+        self._tx_tracker.c_start(clock, timestamp)
+        MarketBase.c_start(self, clock, timestamp)
+
+    cdef c_stop(self, Clock clock):
+        MarketBase.c_stop(self, clock)
+        self._async_scheduler.stop()
+
+    async def start_network(self):
+        if self._order_tracker_task is not None:
+            self._stop_network()
+        self._order_tracker_task = safe_ensure_future(self._order_book_tracker.start())
+        self._trading_rules_polling_task = safe_ensure_future(self._trading_rules_polling_loop())
+        if self._trading_required:
+            self._status_polling_task = safe_ensure_future(self._status_polling_loop())
+            self._user_stream_tracker_task = safe_ensure_future(self._user_stream_tracker.start())
+            self._user_stream_event_listener_task = safe_ensure_future(self._user_stream_event_listener())
+
+    def _stop_network(self):
+        if self._order_tracker_task is not None:
+            self._order_tracker_task.cancel()
```

Comment:
Instead of doing `self._order_tracker_task.cancel()`  - which really has no effect... you should call `self._order_book_tracker.stop()` instead.